cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -161,19 +192,197 @@ const requestCertificate = async function (
   return reqCertResponse.CertificateArn;
 };
 
+const updateHostedZoneRecords = async function (
+  action,
+  options,
+  envName,
+  appName,
+  domainName,
+  envRoute53,
+  appRoute53,
+  envHostedZoneId
+) {
+  const promises = [];
+  for (const option of options) {
+    switch (option.DomainName) {
+      case `${envName}.${appName}.${domainName}`:
+        promises.push(
+          validateDomain({
+            route53: envRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: "",
+            hostedZoneId: envHostedZoneId,
+          })
+        );
+        break;
+      case `${appName}.${domainName}`:
+        promises.push(
+          validateDomain({
+            route53: appRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: `${appName}.${domainName}`,
+          })
+        );
+        break;
+      case domainName:
+        promises.push(
+          validateDomain({
+            route53: appRoute53,
+            record: option.ResourceRecord,
+            action: action,
+            domainName: domainName,
+          })
+        );
+        break;
+    }
+  }
+  return Promise.all(promises);
+};
+
+const deleteHostedZoneRecords = async function (
+  options,
+  envName,
+  appName,
+  domainName,
+  envRoute53,
+  appRoute53,
+  acm,
+  envHostedZoneId
+) {
+  let isLegacyCert = false;
+  // Legacy cert only has two DomainValidationOptions.
+  if (options.length <= 2) {
+    isLegacyCert = true;
+  }
+
+  let isLastOne = false;
+  let certWithDomainCount = 0;
+  let listCertResp = await acm.listCertificates({}).promise();
+  for (const certSummary of listCertResp.CertificateSummaryList || []) {
+    if (certSummary.DomainName === `${envName}.${appName}.${domainName}`) {
+      certWithDomainCount++;
+    }
+  }
+  let nextToken = listCertResp.NextToken;
+  // Though very unlikely, we'll handle the pagination.
+  while (nextToken && certWithDomainCount < 2) {
+    listCertResp = await acm
+      .listCertificates({
+        NextToken: nextToken,
+      })
+      .promise();
+    for (const certSummary of listCertResp.CertificateSummaryList || []) {
+      if (certSummary.DomainName === `${envName}.${appName}.${domainName}`) {
+        certWithDomainCount++;
+      }
+    }
+    nextToken = listCertResp.NextToken;
+  }
+  if (certWithDomainCount < 2) {
+    isLastOne = true;
+  }
```

Comment:
can we define `isLastOne` above the if-statement (defining the variable closest to where its used)?